modules/app/read_write.py

```diff
@@ -11,56 +13,114 @@ def __init__(self) -> None:
         self.numShareableFiles: int = 0
         self.prevShareableFiles: int = 0
 
+        self.numPDFFiles: int = 0
+        self.prevPDFFiles: int = 0
+
         self.dir = Path(self.settings.filesdir).resolve()
         self.textDir = self.dir.joinpath(self.settings.txt_subdir)
         self.pdfDir = self.dir.joinpath(self.settings.pdf_subdir)
 
-    def getFiles(self, path: Path, count_only: bool = False) -> list:
+        self.suffix = self.settings.file_encrypted_suffix
+        self.exceptions = [self.suffix, self.settings.password_file, "_decrypted"]
+
+    class FilesDict(TypedDict):
+        filename: str
+        contents: bytes
+
+    def getFiles(self, path: Path, count_only: bool = False) -> list[FilesDict]:
         """Get all files in a certain directory."""
-        files = []
+        files: list[ReadWrite.FilesDict] = []
 
         if any(path.glob("*")):
             for file in path.glob("*"):
                 if file.is_file():
                     # Do not read the file if count_only is true
-                    contents = None if count_only else file.read_bytes()
+                    if count_only:
+                        contents = b""
+                    elif file.suffix == ".pdf":
+                        contents = file.read_bytes()
+                    else:
+                        contents = file.read_text().encode()
+
                     files.append({"filename": file.name, "contents": contents})
 
         return files
 
-    def removeFiles(self, path: Path) -> None:
-        """Remove all files in a certain directory."""
-        for file in path.glob("*"):
-            if file.is_file():
-                file.unlink()
+    def hasAnyTextFiles(self) -> bool:
+        """Check whether there are any text files, regardless of the type."""
+        files = self.getFiles(self.textDir, True)
+
+        self.numShareableFiles = len(files)  # decide whether to leave it here …
```

Comment:
Should this line be here in the `hasAnyTextFiles` function or in `hasTextFiles`?

Reply:
in hasTextFiles, otherwise it would show the 'share_files.py' even if there are no valid txt files.

Reply:
Alright, I’ll remove the declaration from hasAnyTextFiles then and keep it in hasTextFiles.